client/components/dot-pager/index.jsx

```diff
@@ -128,31 +163,116 @@ export const DotPager = ( {
 		}
 	}, [ numPages ] );
 
+	const getDragPosition = ( event ) => {
+		if ( event.hasOwnProperty( 'clientX' ) ) {
+			return { x: event.clientX, y: event.clientY };
+		}
+
+		if ( event.targetTouches[ 0 ] ) {
+			return { x: event.targetTouches[ 0 ].clientX, y: event.targetTouches[ 0 ].clientY };
+		}
+
+		const touch = event.changedTouches[ 0 ];
+		return { x: touch.clientX, y: touch.clientY };
+	};
+
+	const handleDragStart = ( event ) => {
+		const position = getDragPosition( event );
+		setDragStartData( position );
+		setPagesStyle( { ...pagesStyle, transitionDuration: `0ms` } ); // Set transition Duration to 0 for smooth dragging.
+		setIsDragging( true );
+	};
+
+	const handleDrag = ( event ) => {
+		if ( ! isDragging ) {
+			return;
+		}
+		const dragPosition = getDragPosition( event );
+		const delta = dragPosition.x - dragStartData.x;
+
+		const offset = getOffset( currentPage ) + delta;
+		// Allow for swipe left
+		if ( numPages !== currentPage + 1 && delta < 0 ) {
+			setPagesStyle( {
+				...pagesStyle,
+				transform: `translate3d(${ offset }px, 0px, 0px)`,
+			} );
+		} else if ( currentPage !== 0 && delta > 0 ) {
+			setPagesStyle( {
+				...pagesStyle,
+				transform: `translate3d(${ offset }px, 0px, 0px)`,
+			} );
+		}
+	};
+
+	const handleDragEnd = ( event ) => {
```

Comment:
Something seems off here. If I start to drag from the left side of the slide in the left direction, and stop dragging outside of the slider, it gets stuck and moves with the mouse forever.

Reply:
We probably also need to add a cancel event here.